src/mon.c

```diff
@@ -2928,9 +2928,12 @@ xkilled(
             u.ublessed = 0;
         if (mdat->maligntyp == A_NONE)
             adjalign((int) (ALIGNLIM / 4)); /* BIG bonus */
-    } else if (mtmp->mtame) {
+    } else if ((mtmp->mtame) 
+               && (wep && wep->oartifact != ART_STORMBRINGER)
+               && (u.ualign.type != CHAOTIC)) {
```

Comment:
This will evaluate to `FALSE` as long as the character is chaotic, even if they aren't wielding Stormbringer and haven't been crowned. It will also evaluate to `FALSE` if the character isn't wielding a weapon, regardless of their alignment, since `(wep && wep->oartifact != ART_STORMBRINGER)` is always false if `wep` is 0.

If the goal is not to punish the character as long as they have been "chosen to steal souls" maybe this would work better:
```c
} else if (mtmp->mtame && u.uevent.uhand_of_elbereth != 3) {
```
Or if you want explicitly to exclude Stormbringer:
```c
} else if (mtmp->mtame && !(wep && wep->oartifact == ART_STORMBRINGER)) {
```
You could also combine the two in order to specifically remove the penalty for crowned chaotics wielding Stormbringer:
```c
} else if (mtmp->mtame && !(wep && wep->oartifact == ART_STORMBRINGER
                            && u.uevent.uhand_of_elbereth == 3)) {
```

Reply:
Also, this part only handles killing your own pet, not attacking shopkeepers and guards.  The `else if` right below it is the one that deals with peaceful non-pets.